forge-game/src/main/java/forge/game/card/CardDamageMap.java

```diff
@@ -126,27 +126,29 @@ public void triggerExcessDamage(boolean isCombat, Map<Card, Integer> lethalDamag
 
             int excess = sum - (damaged.getKey().hasBeenDealtDeathtouchDamage() ? 1 : damaged.getValue());
 
+            if (excess <= 0) {
+                continue;
+            }
+
             // also update the DamageHistory, but overwrite previous excess outcomes
             // because Rith, Liberated Primeval cares about who controlled it at this moment
-            lkiCache.get(damaged.getKey().getId()).setHasBeenDealtExcessDamageThisTurn(excess > 0);
-            if (excess > 0) {
-                if (cause != null && cause.hasParam("ExcessSVar")) {
-                    if ((!cause.hasParam("ExcessSVarCondition") || damaged.getKey().isValid(cause.getParam("ExcessSVarCondition"), cause.getActivatingPlayer(), cause.getHostCard(), cause))
-                            && (!cause.hasParam("ExcessSVarTargeted") || damaged.getKey().equals(cause.getTargetCard()))) {
-                        storedExcess += excess;
-                    }
+            lkiCache.get(damaged.getKey().getId()).setHasBeenDealtExcessDamageThisTurn(true);
```

Comment:
like the comment above says the LKI needs to be always updated for _Rith, Liberated Primeval_ to work correctly

Reply:
does inside the loop would ever make sense to call setHasBeenDealtExcessDamageThisTurn(false) ?

Reply:
Rith looks in the past:
> Rith’s ability will trigger as long as an opponent controlled the creature as excess damage was dealt to it, even if the creature is no longer on the battlefield or you control it at the end of the turn.

that means it needs to be refreshed on LKI otherwise
- I control creature and it's dealt excess
- next opponent gains control of it and it's dealt damage again, but nonexcess

Rith would check DamageHistory LKI, see the opponent controlled instance as excess and trigger

Reply:
imo Rith might need to be more fixed, but i will think about something